src/red.ts

```diff
@@ -129,36 +131,55 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
         if (typeof blue === 'function') {
             return getRedFunction(blue);
         }
-        let isBlueArray = false;
-        try {
-            isBlueArray = isArrayOrNotOrThrowForRevoked(blue);
-        } catch {
-            // blue was revoked - but we call createRedProxy to support distortions
-            return createRedProxy(blue);
-        }
-        if (isBlueArray) {
-            return getRedArray(blue);
-        } else if (typeof blue === 'object') {
+        if (typeof blue === 'object') {
             const red: RedValue | undefined = WeakMapGet(blueMap, blue);
-            if (isUndefined(red)) {
-                return createRedProxy(blue);
+            if (!isUndefined(red)) {
+                return red;
+            }
+            let hasDynamicMark: boolean = false;
+            try {
+                hasDynamicMark = hasOwnPropertyCall(blue, LockerLiveValueMarkerSymbol);
+            } catch {}
```

Comment:
Is the try-catch because a proxy `getOwnPropertyDescriptor` trap could throw for the `hasOwnPropertyCall`?

Reply:
yes, I can add a note.